src/services/lease-service/tests/priority-list-service.test.ts

```diff
@@ -457,3 +462,321 @@ describe('parseLeasesForParkingSpaces', () => {
     expect(result).toEqual([])
   })
 })
+
+describe('assignPriorityToApplicantBasedOnRentalRules', () => {
+  it('should throw error if applicant does not belong to the same listing', async () => {
+    const listing = ListingFactory.params({}).build()
```

Comment:
```suggestion
    const listing = ListingFactory.build()
```
Tror inte du behöver params? 🤔